jetbot/bot.py

```diff
@@ -451,47 +498,85 @@ async def channels(ctx: commands.Context):
+                            if not channel:
+                                print("Creating channel... " + channel_config["name"])
+                                channel = await guild.create_text_channel(channel_config["name"], category=category)
+                                new.append(channel)
+
+                            # Set permissions
+                            print("Updating channel... " + channel_config["name"] + " in " + item["category_name"])
+                            for overwrite in channel.overwrites:
+                                await channel.set_permissions(overwrite, overwrite=None)
+                            await channel.edit(slowmode_delay=5, sync_permissions=True)
+
+                    # Handle legacy single-channel configuration
+                    elif item.get("channel_name"):
+                        channel = await self.product_channel(item["channel_name"], item["category_name"])
+                        if not channel:
+                            print("Creating channel... " + item["channel_name"])
+                            channel = await guild.create_text_channel(item["channel_name"], category=category)
+                            new.append(channel)
 
                         # Set permissions
-                        await channel.set_permissions(guild.default_role,
-                                                    send_messages=False,
-                                                    send_messages_in_threads=False,
-                                                    create_private_threads=False,
-                                                    create_public_threads=False)
-                    else:
-                        # Set the topic
-                        product_category = category.name.lower().strip()
-                        product_emoji = await self.product_emoji(item["emoji_name"])
-                        title = title_map[product_category] if product_category in title_map else default_title
-                        title = title.format(item["name"])
-                        title = "{0} {1} {0}".format(product_emoji, title) if product_emoji else title
-                        if channel.topic != title:
-                            await channel.edit(topic=title)
+                        print("Updating channel... " + item["channel_name"] + " in " + item["category_name"])
+                        for overwrite in channel.overwrites:
+                            await channel.set_permissions(overwrite, overwrite=None)
+                        await channel.edit(slowmode_delay=5, sync_permissions=True)
+
+                    channels_to_update = []
+                    if "channels" in item:
+                        channels_to_update.extend(await self.product_channels(item))
+                    elif item.get("channel_name"):
+                        channel = await self.product_channel(item["channel_name"], item["category_name"])
```

Comment:
Again, a similar thought here, can we condense down into a single branch of logic instead of two?